HoloStorageAccessor/internal/holostorageaccessor/api_default_test.go

```diff
@@ -0,0 +1,1470 @@
+			wantStatus: 400,
+		},
+		"single_found_author": {
+			inUrl:      "/api/v1/authors?aids=a1000",
+			inMethod:   http.MethodGet,
+			wantStatus: 200,
+			wantBody: map[string]Author{
+				"a1000": Author{
+					Aid: "a1000",
+					Name: &PersonName{
+						Family: "Campbell",
+						Full:   "Roy Campbell",
+						Given:  "Roy",
+						Title:  "Mr",
+					},
+				},
+			},
+		},
+		"single_missing_author": {
+			inUrl:      "/api/v1/authors?aids=a3000",
+			inMethod:   http.MethodGet,
+			wantStatus: 200,
+			wantBody: map[string]Author{
+				"a3000": Author{},
+			},
+		},
+		"found_author": {
+			inUrl:      "/api/v1/authors?aids=a1000,a2000,a3000",
+			inMethod:   http.MethodGet,
+			wantStatus: 200,
+			wantBody: map[string]Author{
+				"a1000": Author{
+					Aid: "a1000",
+					Name: &PersonName{
+						Family: "Campbell",
+						Full:   "Roy Campbell",
+						Given:  "Roy",
+						Title:  "Mr",
+					},
+				},
+				"a2000": Author{
+					Aid: "a2000",
+					Name: &PersonName{
+						Family: "Sawyer",
+						Full:   "Tom Sawyer",
+						Given:  "Tom",
+						Title:  "Mr",
+					},
+				},
+				"a3000": Author{},
+			},
+		},
+	}
+
+	// Start a local HTTP server and Router
+	ts := setupTestServer()
+	defer ts.Close()
+	router := NewRouter(AccessorConfig{FhirURL: ts.URL})
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			diff := ""
+			req, _ := http.NewRequest(tc.inMethod, tc.inUrl, nil)
+			w := httptest.NewRecorder()
+			router.ServeHTTP(w, req)
+
+			if tc.wantStatus != w.Code {
+				t.Fatalf("Error return code: Want %d, got %d", tc.wantStatus, w.Code)
+			}
+
+			if tc.wantBody != nil {
+				var authorData map[string]Author
+				err := json.Unmarshal(w.Body.Bytes(), &authorData)
+				if err != nil {
+					t.Fatalf("Unmarshal Author error: %s", err.Error())
+				}
+				diff = cmp.Diff(tc.wantBody, authorData)
+			} else if (tc.wantErr != Error{}) {
+				var errorData Error
+				err := json.Unmarshal(w.Body.Bytes(), &errorData)
+				if err != nil {
+					t.Logf(string(w.Body.Bytes()))
+					t.Fatalf("Unmarshal Error error: %s", err.Error())
+				}
+				diff = cmp.Diff(tc.wantErr, errorData)
+			}
+
+			if diff != "" {
+				t.Fatalf(diff)
+			}
+		})
+	}
+
+}
+
+func TestAuthorsAidGet(t *testing.T) {
+	type test struct {
+		wantStatus int
+		wantBody   Author
+		wantErr    Error
+		inUrl      string
+		inMethod   string
+	}
+
+	tests := map[string]test{
+		"found_author": {
+			inUrl:      "/api/v1/authors/a1000",
+			inMethod:   http.MethodGet,
+			wantStatus: 200,
+			wantBody: Author{
+				Aid: "a1000",
+				Name: &PersonName{
+					Family: "Campbell",
+					Full:   "Roy Campbell",
+					Given:  "Roy",
+					Title:  "Mr",
+				},
+			},
+		},
+		"missing_author": {
+			inUrl:      "/api/v1/authors/a3000",
+			inMethod:   http.MethodGet,
+			wantStatus: 404,
+			wantErr: Error{
+				ErrorCode:    "404",
+				ErrorMessage: `id 'a3000' cannot be found`,
+			},
+		},
+	}
+
+	// Start a local HTTP server and Router
```

Comment:
I don't understand why you do this eight times?

Reply:
Changed it such that there's a `TestMain()` function that does an initial setup before running all tests.